### PR TITLE
vSphere: support for VM Groups

### LINKED ISSUE #1846: vSphere: Support for VM Groups
As a KKP user, I'd like to be able to assign machine deployments to specific VM groups to control their placement in a deterministic manner.

## Acceptance Criteria

- [ ] Assign VM/Host group to VMs

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds support for adding VMs to pre-defined VM groups. This is useful for VM placement as host rules can be created for VM groups in vSphere.

**Which issue(s) this PR fixes**:

Fixes #1846

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
vSphere: Support for assigning VMs to VM groups. 
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
vSphere tests are supposed to be fixed in https://github.com/kubermatic/machine-controller/pull/1844. So we'll need to override them in this PR so that I can continue with the implementation in KKP and dashboard.

File: pkg/cloudprovider/provider/vsphere/vmgroup.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+			vmRefs = append(vmRefs, vm.Reference())
+		}
+	}
+
+	var vmRefsToAdd []types.ManagedObjectReference
+	for _, vm := range vmRefs {
+		for _, existingVM := range vmGroup.Vm {
+			if existingVM.Value == vm.Value {
+				log.Debugf("VM %s already in VM group %s", machine.Name, config.VMGroup)
+				continue
```
Comment (mohamed-rafraf) [inline pkg/cloudprovider/provider/vsphere/vmgroup.go:61]: If we find the VM in the VM group we can break the inner loop instead.

File: pkg/cloudprovider/provider/vsphere/vmgroup.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			}
+		}
+		vmRefsToAdd = append(vmRefsToAdd, vm)
```
Comment (mohamed-rafraf) [inline pkg/cloudprovider/provider/vsphere/vmgroup.go:64]: In this case you append very VM to VMRefsToAdd regardless of whether it was already in the group or not

Comment (ahmedwaleedmalik) [reply pkg/cloudprovider/provider/vsphere/vmgroup.go:64]: Yeah I mistakenly added continue instead of break :)

**Pull request merged**